src/Button/index.jsx

```diff
@@ -1,6 +1,58 @@
-import Button from 'react-bootstrap/Button';
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import ButtonBase from 'react-bootstrap/Button';
+
+import { useHandleLogClick, useTrackComponentOnMount } from '../hooks/analytics';
+
 import ButtonDeprecated from './deprecated';
 
+const Button = ({
+  children,
+  analyticEvents,
+  analyticDispatchers,
+  ...attrs
+}) => {
+  useTrackComponentOnMount('edx.ui.paragon.Button.mounted', attrs);
+
+  const ref = useRef();
```

Comment:
TODO: i believe our components should always use `React.forwardRef` so consumers can pass in a custom ref to use (should confirm that react-bootstrap also does this).